From 143b0493cddbf4885a93e54f6253f51f74953a96 Mon Sep 17 00:00:00 2001
From: zhangmeng <zhangmeng.kevin@linux.spacemit.com>
Date: Mon, 19 May 2025 15:22:29 +0800
Subject: [PATCH 13/43] riscv: k1: dts: add spi support for spacemit k1

community inclusion
category: feature
bugzilla: https://github.com/RVCK-Project/rvck-olk/issues/53
Reference: https://gitee.com/bianbu-linux/linux-6.6/commit/c57a5517737787dcc26af4a868f5559608e57f5f

--------------------------------

there are 4 spi controller in k1, but the spi2 is reserved
for TEE secure world, so only spi0/1/3 used for linux

Signed-off-by: zhangmeng <zhangmeng.kevin@linux.spacemit.com>
---
 arch/riscv/boot/dts/spacemit/k1-x.dtsi | 57 ++++++++++++++++++++++++++
 1 file changed, 57 insertions(+)

diff --git a/arch/riscv/boot/dts/spacemit/k1-x.dtsi b/arch/riscv/boot/dts/spacemit/k1-x.dtsi
index 6a08c44b171b9..b9fb1751f0bad 100644
--- a/arch/riscv/boot/dts/spacemit/k1-x.dtsi
+++ b/arch/riscv/boot/dts/spacemit/k1-x.dtsi
@@ -827,6 +827,63 @@ i2c8: i2c@d401d800 {
 			status = "disabled";
 		};
 
+		spi0: spi@d4026000 {
+			compatible = "spacemit,k1-spi";
+			reg = <0x0 0xd4026000 0x0 0x30>;
+			k1,spi-id = <0>;
+			k1,spi-clock-rate = <26000000>;
+			dmas = <&pdma0 DMA_SSPA0_RX 1
+				&pdma0 DMA_SSPA0_TX 1>;
+			dma-names = "rx", "tx";
+			interrupt-parent = <&intc>;
+			interrupts = <56>;
+			clocks = <&ccu CLK_SSPA0>;
+			resets = <&reset RESET_SSPA0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			interconnects = <&dram_range0>;
+			interconnect-names = "dma-mem";
+			status = "disabled";
+		};
+
+		spi1: spi@d4026800 {
+			compatible = "spacemit,k1-spi";
+			reg = <0x0 0xd4026800 0x0 0x30>;
+			k1,spi-id = <1>;
+			k1,spi-clock-rate = <26000000>;
+			dmas = <&pdma0 DMA_SSPA1_RX 1
+				&pdma0 DMA_SSPA1_TX 1>;
+			dma-names = "rx", "tx";
+			interrupt-parent = <&intc>;
+			interrupts = <57>;
+			clocks = <&ccu CLK_SSPA1>;
+			resets = <&reset RESET_SSPA1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			interconnects = <&dram_range0>;
+			interconnect-names = "dma-mem";
+			status = "disabled";
+		};
+
+		spi3: spi@d401c000 {
+			compatible = "spacemit,k1-spi";
+			reg = <0x0 0xd401c000 0x0 0x30>;
+			k1,spi-id = <3>;
+			k1,spi-clock-rate = <26000000>;
+			dmas = <&pdma0 DMA_SSP3_RX 1
+				&pdma0 DMA_SSP3_TX 1>;
+			dma-names = "rx", "tx";
+			interrupt-parent = <&intc>;
+			interrupts = <55>;
+			clocks = <&ccu CLK_SSP3>;
+			resets = <&reset RESET_SSP3>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			interconnects = <&dram_range0>;
+			interconnect-names = "dma-mem";
+			status = "disabled";
+		};
+
 		gpio: gpio@d4019000 {
 			compatible = "spacemit,k1x-gpio";
 			reg = <0x0 0xd4019000 0x0 0x800>;
-- 
2.51.0

